index.js

```diff
@@ -406,7 +402,14 @@ class Hyperbee extends ReadyResource {
   }
 
   async _open () {
-    if (this.core.opened === false) await this.core.ready()
+    await this.core.ready()
+
+    if (this.hasExtension) this.extension = this.extension || Extension.register(this)
+
+    if (this._watchers) {
```

Comment:
What's the context for the change? I wonder if with this change there's some edge case where the core's append/truncate triggers before the listeners below are set, causing the watchers not to trigger